test/e2e_node/kubelet_authz_test.go

```diff
@@ -0,0 +1,125 @@
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/kubernetes/pkg/cluster/ports"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eauth "k8s.io/kubernetes/test/e2e/framework/auth"
+)
+
+var _ = SIGDescribe("Kubelet Authz", func() {
+	f := framework.NewDefaultFramework("kubelet-authz-test")
+	ginkgo.Context("when calling kubelet API", func() {
+		ginkgo.It("check /healthz enpoint is accessible via nodes/healthz RBAC", func(ctx context.Context) {
+			sc := runKubeletAuthzTest(ctx, f, "healthz", "healthz")
+			gomega.Expect(sc).To(gomega.Equal(http.StatusOK))
+		})
+		ginkgo.It("check /healthz enpoint is accessible via nodes/proxy RBAC", func(ctx context.Context) {
+			sc := runKubeletAuthzTest(ctx, f, "healthz", "proxy")
+			gomega.Expect(sc).To(gomega.Equal(http.StatusOK))
+		})
+		ginkgo.It("check /healthz enpoint is not accessible via nodes/configz RBAC", func(ctx context.Context) {
+			sc := runKubeletAuthzTest(ctx, f, "healthz", "configz")
+			gomega.Expect(sc).To(gomega.Equal(http.StatusUnauthorized))
+		})
+	})
+})
+
+func runKubeletAuthzTest(ctx context.Context, f *framework.Framework, endpoint, authzSubresource string) int {
+	ns := f.Namespace.Name
+	saName := authzSubresource
+	crName := authzSubresource
+	verb := "get"
+	resource := "nodes"
+	_, err := f.ClientSet.CoreV1().ServiceAccounts(ns).Create(ctx, &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: ns,
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	_, err = f.ClientSet.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{verb},
+				Resources: []string{resource + "/" + authzSubresource},
+			},
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	subject := rbacv1.Subject{
+		Kind:      rbacv1.ServiceAccountKind,
+		Namespace: ns,
+		Name:      saName,
+	}
+
+	err = e2eauth.BindClusterRole(ctx, f.ClientSet.RbacV1(), crName, ns, subject)
+	framework.ExpectNoError(err)
+
+	err = e2eauth.WaitForAuthorizationUpdate(ctx, f.ClientSet.AuthorizationV1(),
+		serviceaccount.MakeUsername(ns, saName),
+		&authorizationv1.ResourceAttributes{
+			Namespace:   ns,
+			Verb:        verb,
+			Resource:    resource,
+			Subresource: authzSubresource,
+		},
+		true,
+	)
+	framework.ExpectNoError(err)
+
+	tr, err := f.ClientSet.CoreV1().ServiceAccounts(ns).CreateToken(ctx, saName, &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	resp, err := healthCheck(fmt.Sprintf("http://127.0.0.1:%d/%s", ports.KubeletPort, endpoint), tr.Status.Token)
```

Comment:
Shouldn't this be `https`?

Reply:
I'll add this in a separate follow up PR.